File: DataStructures/Queue/QueueLinkedList.java
public class QueueLinkedList {
    private LinkedList linkedList;

    public QueueLinkedList() {
        this.linkedList = new LinkedList();
    }

    // Insere no final da fila
    public void enqueue(int elemento) {
        linkedList.add(elemento);
    }

    // Remove do início da fila
    public void dequeue() {
        if (!isEmpty()) {
            linkedList.delete(linkedList.getFirst().getValue());
        } else {
            System.out.println("A fila está vazia.");
        }
    }

    // Mostra todos os elementos da fila
    public void show() {
        for (int i = 0; i < linkedList.getSize(); i++) {
            System.out.println(linkedList.get(i).getValue());
        }
    }

    public boolean isEmpty() {
        return linkedList.getSize() == 0;
    }
}

File: DataStructures/Queue/QueueLinkedList.rb
class QueueLinkedList
  def initialize
    @linked_list = LinkedList.new
  end

  def enqueue(elemento)
    @linked_list.add(elemento)
  end

  def dequeue
    if is_empty?
      puts "A fila está vazia."
    else
      @linked_list.delete(@linked_list.first.value)
    end
  end

  def show
    current = @linked_list.first
    while current
      puts current.value
      current = current.next
    end
  end

  def is_empty?
    @linked_list.size == 0
  end
end
